com.microsoft.codepush.react.CodePushMalformedDataException Unable to parse contents of [..]/codepush.json

Hi,

this is the crash report I've got from crash-analytics service. I wasn't able to reproduce it myself however whatever happens the codepush should not crash, it should safely fallback to original bundle if error happened during it's initialization.

### Expected Behavior

CodePush should not crash the app. It should safely fallback to original bundle.

### Actual Behavior

App crashes.

```
java.lang.RuntimeException An error occurred while executing doInBackground() 
    AsyncTask.java:318 android.os.AsyncTask$3.done
    FutureTask.java:354 java.util.concurrent.FutureTask.finishCompletion
    FutureTask.java:223 java.util.concurrent.FutureTask.setException
    FutureTask.java:242 java.util.concurrent.FutureTask.run
    ThreadPoolExecutor.java:1133 java.util.concurrent.ThreadPoolExecutor.runWorker
    ThreadPoolExecutor.java:607 java.util.concurrent.ThreadPoolExecutor$Worker.run
    Thread.java:761 java.lang.Thread.run


Caused by: com.microsoft.codepush.react.CodePushMalformedDataException Unable to parse contents of /data/user/0/my.package.name/files/CodePush/codepush.json, the file may be corrupted. 
    CodePushUtils.java:198 com.microsoft.codepush.react.CodePushUtils.getJsonObjectFromFile
    CodePushUpdateManager.java:55 com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackageInfo
    CodePushUpdateManager.java:105 com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackageHash
    CodePushUpdateManager.java:115 com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackage
    CodePushNativeModule.java:295 com.microsoft.codepush.react.CodePushNativeModule$4.doInBackground
    CodePushNativeModule.java:292 com.microsoft.codepush.react.CodePushNativeModule$4.doInBackground
    AsyncTask.java:304 android.os.AsyncTask$2.call
    FutureTask.java:237 java.util.concurrent.FutureTask.run
    ThreadPoolExecutor.java:1133 java.util.concurrent.ThreadPoolExecutor.runWorker
    ThreadPoolExecutor.java:607 java.util.concurrent.ThreadPoolExecutor$Worker.run
    Thread.java:761 java.lang.Thread.run


Caused by: org.json.JSONException End of input at character 0 of  
    JSONTokener.java:449 org.json.JSONTokener.syntaxError
    JSONTokener.java:97 org.json.JSONTokener.nextValue
    JSONObject.java:156 org.json.JSONObject.<init>
    JSONObject.java:173 org.json.JSONObject.<init>
    CodePushUtils.java:195 com.microsoft.codepush.react.CodePushUtils.getJsonObjectFromFile
    CodePushUpdateManager.java:55 com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackageInfo
    CodePushUpdateManager.java:105 com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackageHash
    CodePushUpdateManager.java:115 com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackage
    CodePushNativeModule.java:295 com.microsoft.codepush.react.CodePushNativeModule$4.doInBackground
    CodePushNativeModule.java:292 com.microsoft.codepush.react.CodePushNativeModule$4.doInBackground
    AsyncTask.java:304 android.os.AsyncTask$2.call
    FutureTask.java:237 java.util.concurrent.FutureTask.run
    ThreadPoolExecutor.java:1133 java.util.concurrent.ThreadPoolExecutor.runWorker
    ThreadPoolExecutor.java:607 java.util.concurrent.ThreadPoolExecutor$Worker.run
    Thread.java:761 java.lang.Thread.run
```

### Reproducible Demo

* If you can't reproduce the bug on it, provide us as much info as possible about your project

I do have this 
`apply from: "../../node_modules/react-native-code-push/android/codepush.gradle"` line in `build.gradle`

### Environment

* react-native-code-push version: 5.2.1
* react-native version: 0.50.3
* iOS/Android/Windows version: android 7.0
* Does this reproduce on a debug build or release build?
* Does this reproduce on a simulator, or only on a physical device?
Happened dozen of times on release build on physical device.

Also happening here, however it seems to only occur in a specific device (Twist S520).

```
Positivo Twist (S520), Android 6.0
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2583)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2665)
  at android.app.ActivityThread.-wrap11 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1499)
  at android.os.Handler.dispatchMessage (Handler.java:111)
  at android.os.Looper.loop (Looper.java:207)
  at android.app.ActivityThread.main (ActivityThread.java:5765)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:789)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:679)
Caused by: com.microsoft.codepush.react.CodePushMalformedDataException: 
  at com.microsoft.codepush.react.CodePushUtils.getJsonObjectFromFile (CodePushUtils.java:198)
  at com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackageInfo (CodePushUpdateManager.java:55)
  at com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackageHash (CodePushUpdateManager.java:105)
  at com.microsoft.codepush.react.CodePushUpdateManager.getCurrentPackage (CodePushUpdateManager.java:115)
  at com.microsoft.codepush.react.CodePush.initializeUpdateAfterRestart (CodePush.java:254)
  at com.microsoft.codepush.react.CodePush.<init> (CodePush.java:82)
  at com.educarebox.app.MainApplication$1.getPackages (MainApplication.java:47)
  at com.facebook.react.ReactNativeHost.createReactInstanceManager (ReactNativeHost.java:81)
  at com.facebook.react.ReactNativeHost.getReactInstanceManager (ReactNativeHost.java:45)
  at com.facebook.react.ReactActivityDelegate.loadApp (ReactActivityDelegate.java:88)
  at com.facebook.react.ReactActivityDelegate.onCreate (ReactActivityDelegate.java:77)
  at com.facebook.react.ReactActivity.onCreate (ReactActivity.java:54)
  at android.app.Activity.performCreate (Activity.java:6301)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1113)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2530)
Caused by: org.json.JSONException: 
  at org.json.JSONTokener.syntaxError (JSONTokener.java:449)
  at org.json.JSONTokener.nextValue (JSONTokener.java:97)
  at org.json.JSONObject.<init> (JSONObject.java:156)
  at org.json.JSONObject.<init> (JSONObject.java:173)
  at com.microsoft.codepush.react.CodePushUtils.getJsonObjectFromFile (CodePushUtils.java:195)
```

--- a/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java
+++ b/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java
@@ -213,7 +213,15 @@ public class CodePush implements ReactPackage {
         this.mAssetsBundleFileName = assetsBundleFileName;
         String binaryJsBundleUrl = CodePushConstants.ASSETS_BUNDLE_PREFIX + assetsBundleFileName;
 
-        String packageFilePath = mUpdateManager.getCurrentPackageBundlePath(this.mAssetsBundleFileName);
+        String packageFilePath = null;
+        try {
+            packageFilePath = mUpdateManager.getCurrentPackageBundlePath(this.mAssetsBundleFileName);
+        } catch (CodePushMalformedDataException e) {
+            // We need to recover the app in case 'codepush.json' is corrupted
+            CodePushUtils.log(e.getMessage());
+            clearUpdates();
+        }
+
         if (packageFilePath == null) {
             // There has not been any downloaded updates.
             CodePushUtils.logBundleUrl(binaryJsBundleUrl);

--- a/android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java
+++ b/android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java
@@ -349,6 +349,11 @@ public class CodePushNativeModule extends ReactContextBaseJavaModule {
                         CodePushUtils.setJSONValueForKey(currentPackage, "isPending", currentUpdateIsPending);
                         promise.resolve(CodePushUtils.convertJsonObjectToWritable(currentPackage));
                     }
+                } catch (CodePushMalformedDataException e) {
+                    // We need to recover the app in case 'codepush.json' is corrupted
+                    CodePushUtils.log(e.getMessage());
+                    clearUpdates();
+                    promise.resolve(null);
                 } catch(CodePushUnknownException e) {
                     CodePushUtils.log(e);
                     promise.reject(e);

